core/storage/app_feedback_report/gae_models.py

```diff
@@ -0,0 +1,659 @@
+            platform: str. The platform the user is the report from.
+            submitted_on_sec: float. The timestamp that the report was submitted
+                on, in seconds since epoch (in UTC).
+
+        Returns:
+            str. The generated ID for this entity using platform,
+            submitted_on_sec, and a random string, of the form
+            '[platform].[submitted_on_sec].[random hash]'.
+        """
+        for _ in python_utils.RANGE(base_models.MAX_RETRIES):
+            random_hash = utils.convert_to_hash(
+                python_utils.UNICODE(
+                    utils.get_random_int(base_models.RAND_RANGE)),
+                base_models.ID_LENGTH)
+            new_id = '%s.%s.%s' % (platform, submitted_on_sec, random_hash)
+            if not cls.get_by_id(new_id):
+                return new_id
+        raise Exception(
+            'The id generator for AppFeedbackReportModel is producing too '
+            'many collisions.')
+
+    @classmethod
+    def scrub_report(cls, report_id, scrubbed_by):
+        """Scrubs the instance of AppFeedbackReportModel with given ID, removing
+        any user-entered input in the entity.
+
+        Args:
+            report_id: str. The id of the model entity to scrub.
+            scrubbed_by: str. The id of the user or cron job that is intiating
+                scrubbing this report.
+        """
+        cls._scrub_report_in_transaction(report_id, scrubbed_by)
+
+    @classmethod
+    @transaction_services.run_in_transaction_wrapper
+    def _scrub_report_in_transaction(cls, report_id, scrubbed_by):
+        """See scrub_report for general documentaion of what this method does.
+        It's only safe to call this method from within a transaction.
+
+        Args:
+            report_id: str. The id of the model entity to scrub.
+            scrubbed_by: str. The id of the user or cron job that is intiating
+                scrubbing this report.
+        """
+        report_entity = cls.get_by_id(report_id)
+        if not report_entity:
+            raise Exception(
+                'The AppFeedbackReportModel trying to be scrubbed does not '
+                'exist.')
+        if report_entity.platform == PLATFORM_CHOICE_ANDROID:
+            scrubbed_report_info = cls._scrub_report_info(
+                report_entity.android_report_info)
+            report_entity.android_report_info = scrubbed_report_info
+        else:
+            scrubbed_report_info = cls._scrub_report_info(
+                report_entity.web_report_info)
+            report_entity.web_report_info = scrubbed_report_info
+        report_entity.scrubbed_by = scrubbed_by
+        report_entity.update_timestamps()
+        report_entity.put()
+
+    @staticmethod
+    def _scrub_report_info(report_info_dict):
+        """Scrubs the dictionary of any fields that contains input directly from
+        the user.
+
+        Args: report_info_dict: dict. The info dict collected in a report.
```

Comment:
This is not the correct docstring format. Please follow the existing codebase conventions.

@Hudda @sajalasati @DubeySandeep Shouldn't this be caught by the linter?

Reply:
@seanlip I did an inspection, and for now I think we don't have this lint check yet. But will need to do more analysis before I can confirm if it's a bug/missing check. I have created #12349 to track this. Thanks for reporting this bug!

Reply:
Fixed, done.